### PR TITLE
Simplify typing

### PR DESCRIPTION
**Pull request by felixfontein**
Currently contains #93, will rebase once that's merged.

### CODE REVIEW

File: src/antsibull_changelog/changes.py [hunk 1]
```diff
@@ -10,12 +10,14 @@
...
 
+from __future__ import annotations
+
 import abc
 import collections
 import datetime
 import os
 
-from typing import Any, Callable, Dict, List, Optional, Set, cast
+from typing import Any, Callable, cast
```
Comment (gotmax23) [inline src/antsibull_changelog/changes.py:20]: `collections.abc.Callable`

File: src/antsibull_changelog/cli.py [hunk 2]
```diff
@@ -8,13 +8,15 @@
...
+from __future__ import annotations
+
 import argparse
 import datetime
 import os
 import sys
 import traceback
 
-from typing import Any, Callable, cast, List, Optional, Tuple
+from typing import Any, Callable, cast, List
```
Comment (gotmax23) [inline src/antsibull_changelog/cli.py:19]: `collections.abc.Callable` should be used instead.

``` console
$ rg 'from typing import.*Callable'
stubs/tomli.pyi:from typing import Callable, Any
src/antsibull_changelog/utils.py:from typing import Any, Callable
src/antsibull_changelog/cli.py:from typing import Any, Callable, cast, List
src/antsibull_changelog/changes.py:from typing import Any, Callable, cast
tests/functional/fixtures.py:from typing import Any, Callable
```

Comment (gotmax23) [reply src/antsibull_changelog/cli.py:19]: `typing.List` is used along with `cast()` in this file. Those can use `list` as well. That is available at runtime on Python 3.9 and above.

File: src/antsibull_changelog/utils.py [hunk 3]
```diff
@@ -9,9 +9,12 @@
 Utility functions.
 """
 
+from __future__ import annotations
+
 import re
 
-from typing import Any, Callable, Collection, List, Optional, Tuple
+from collections.abc import Collection
+from typing import Any, Callable
```
Comment (gotmax23) [inline src/antsibull_changelog/utils.py:17]: `collections.abc.Callable`

File: tests/functional/fixtures.py [hunk 4]
```diff
@@ -7,15 +7,18 @@
...
 import io
 import json
 import os
 import pathlib
 import textwrap
 
+from collections.abc import Mapping
 from contextlib import redirect_stderr, redirect_stdout
-from typing import Any, Callable, Dict, List, Mapping, Optional, Set, Tuple, Union
+from typing import Any, Callable
```
Comment (gotmax23) [inline tests/functional/fixtures.py:21]: `collections.abc.Callable`

**Comment by gotmax23** [comment]
There are some remaining usages in `stubs`. Not sure if you want to leave those

``` console
$ rg -e 'import typing' -e 'from typing' stubs
stubs/tomli.pyi:from typing import Callable, Any
stubs/rstcheck.pyi:from typing import List, Optional, Tuple, Union
stubs/rstcheck_core/config.pyi:from typing import List, Optional, Tuple, Union
stubs/rstcheck_core/types.pyi:from typing import Literal, Union
stubs/rstcheck_core/types.pyi:    from typing import TypedDict
stubs/rstcheck_core/types.pyi:    from typing_extensions import TypedDict
stubs/rstcheck_core/checker.pyi:from typing import List, Optional, Tuple, Union
stubs/ansible/constants.pyi:from typing import Tuple
```

File: tests/functional/fixtures.py (hunk 4 above)
Comment (felixfontein) [reply tests/functional/fixtures.py]: 0f3295ce0448eb53c8b795f74597202a3fb6df65.

File: src/antsibull_changelog/utils.py (hunk 3 above)
Comment (felixfontein) [reply src/antsibull_changelog/utils.py]: 0f3295ce0448eb53c8b795f74597202a3fb6df65.

File: src/antsibull_changelog/cli.py (hunk 2 above)
Comment (felixfontein) [reply src/antsibull_changelog/cli.py]: 0f3295ce0448eb53c8b795f74597202a3fb6df65.

File: src/antsibull_changelog/changes.py (hunk 1 above)
Comment (felixfontein) [reply src/antsibull_changelog/changes.py]: 0f3295ce0448eb53c8b795f74597202a3fb6df65.

**Review by gotmax23 — APPROVED** [review]

**Comment by gotmax23** [comment]
Thanks!

**Comment by felixfontein** [comment]
@gotmax23 thanks for reviewing and merging!

**Pull request merged**